### PR TITLE
feat: add slider component

### PR DESCRIPTION
**Pull request by pkc918**

### CODE REVIEW

**Comment by baiwusanyu-c** [comment]
This PR contains the content of the Switch component, if there is a change please submit a new PR, if not, then you should create a new branch from the upstream main branch to write the slider component

File: preset/src/shortcuts/slider.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const sliderShortcuts: Record<string, string> = {
+    // select
```
Comment (baiwusanyu-c) [inline preset/src/shortcuts/slider.ts:2]: `preset/src/shortcuts/common.ts` contains shortcuts for some class combinations, you can use them, such as `flex items-center justify-center -> fcc`

File: docs/components/KSlider.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+## Render slider
+
+<demo src="../../../../example/slider/render.svelte" github="Slider"></demo>
+
+## Slider Props
+
+| Name     | Type      | Default | Description                |
+|----------|-----------|---------|----------------------------|
+| value    | `number`  | `0`     | Binding value              |
+| min      | `number`  | `0`     | Minimum value              |
```
Comment (baiwusanyu-c) [inline docs/components/KSlider.md:47]: It is recommended to add `min`, `max`, `step` document use cases

File: components/Slider/src/index.svelte [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	let startPosition: number;
+	let newPosition: number;
+
+	$:percentage = `${(value - min) / (max - min) * 100}%`;
+
+	// element
+	let sliderRunwayRef: null | HTMLElement = null;
+	// updateValue
+	const dispatch = createEventDispatcher();
+	const handleMouseEnter = () => {
```
Comment (baiwusanyu-c) [inline components/Slider/src/index.svelte:24]: Seems like the component isn't finished yet?

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

**Comment by baiwusanyu-c** [comment]
The slider component needs to add unit tests

File: preset/src/shortcuts/slider.ts (hunk 1 above)
Comment (pkc918) [reply preset/src/shortcuts/slider.ts:2]: oh my oversight

File: components/Slider/src/index.svelte (hunk 3 above)
Comment (pkc918) [reply components/Slider/src/index.svelte:24]: You're absolutely right. The Slider conponent isn't fully finished yet. It's currently a basic version.

Comment (pkc918) [reply components/Slider/src/index.svelte:24]: You're absolutely right. The component isn't fully finished yet. It's currently a basic version

**Pull request closed without merging**